demo/docker-compose.yml

```diff
@@ -1,36 +1,42 @@
-version: '3.7'
+version: '3.8'
+volumes:
+  pgdata:
+    driver: local
 services:
   postgres:
-    build:
-      context: postgres
-      dockerfile: Dockerfile
-    shm_size: 256MB
+    image: postgres:15beta2
+    # https://dba.stackexchange.com/q/275378/46472
+    shm_size: 3GB
     volumes:
+      - type: volume
+        source: pgdata
+        target: /var/lib/postgresql/data/
       - type: bind
-        source: ./postgres/db
-        target: /db
-    sysctls:
-      net.core.somaxconn: 256
-      net.ipv4.tcp_syncookies: 0
-      # To use all of 2 gigabytes
-      kernel.shmmax: 1055092736
-      kernel.shmall: 257591
-    privileged: true
+        source: ./postgres/init
+        target: /docker-entrypoint-initdb.d
     environment:
-      PG_TRUST_LOCALNET: 'true'
-    command: tail -f /dev/null
+      POSTGRES_DB: fhirdb
+      POSTGRES_USER: fhiradmin
+      POSTGRES_PASSWORD: change-password
+      PGDATA: /var/lib/postgresql/data/pgdata
     healthcheck:
-      start_period: 21s
       test: ["CMD-SHELL", "pg_isready -d fhirdb -U fhiradmin"]
-      interval: 10s
-      timeout: 5s
+      interval: 5s
+      timeout: 4s
       retries: 3
+    command: "
+      -c shared_buffers=2GB
```

Comment:
make sure the container is also allocated sufficient memory

Reply:
do you know what happens if shared_buffers is set higher than the alotted memory?

Reply:
I could drop this back down to 1 GB like I have in our CI pipeline, but I was trying a slightly higher value since we used to set it to 4 GB.

Reply:
The database will halt if it can't allocate sufficient shared memory. The default configuration for PostgreSQL is to use mmap to attach to anonymous shared memory using /dev/shm which on Linux is mounted as a tmpfs and is sized by specifying the `--shm-size` argument to docker run (or docker-compose equivalent).

If you bash into the container you can see the mount:
```
docker exec -ti your-container-name bash
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
bash-5.0# mount | grep shm
shm on /dev/shm type tmpfs (rw,context="system_u:object_r:container_file_t:s0:c640,c766",nosuid,nodev,noexec,relatime,size=3906252k,uid=1000,gid=1001)
```
In this case, the container was started with `--shm-size=4g` which corresponds to `size=3906252k` (4000/1024).

However, it looks like PostgreSQL isn't using /dev/shm for its shared buffers. It uses shared memory segments from /dev/shm during parallel query (think our large search queries). The amount of memory it uses will depend on the number of parallel query workers and the work_mem.

Reply:
In terms of accommodating shared_buffers, it depends if the container has been configured with a memory limit. If not, then it depends how much (virtual) memory is available on the host.